include/odp/api/spec/crypto.h

```diff
@@ -688,8 +722,11 @@ typedef struct odp_crypto_session_param_t {
 	/** Output pool
 	 *
 	 *  When the output packet is not specified during the call to
-	 *  crypto operation, the output packet will be allocated
-	 *  from this pool.
+	 *  crypto operation in the legacy operation type, the output
+	 *  packet will be allocated from this pool.
+	 *
+	 *  In ODP_CRYPTO_OP_TYPE_COPY operation types this must be set to
```

Comment:
When op_type is 'ODP_CRYPTO_OP_TYPE_COPY' which is the pool to be used for packet allocation?

Reply:
It is the same pool as the input packet. See the documentation of odp_crypto_op() which says it.

Reply:
Thanks for the confirmation. So, the implementation is expected to get the pool from the input packet in datapath and then allocate new buffer?

Is the expectation to use odp_packet_pool() for getting the pool?

Reply:
Conceptually, the implementation checks the pool of the input packet (e.g. using odp_packet_pool), allocates the output packet from that pool and then copies all data and metadata from the input packet to the output packet. Finally the implementation frees the input packet (except if the application has requested the input packet to be preserved, which I think is not the typical case). In practice the implementation could simply reuse the input packet (if the input packet is not to be preserved) and the end result would be the same. But the API does not promise that the input packet gets reused, so it is up to the implementation to reuse it or not.

Reply:
When input packet needs to be preserved, implementation is expected to do odp_packet_pool() and allocate from the same, right?

Reply:
Yes (except, of course, the implementation can figure out the pool by any internal means, without calling ODP API functions).